src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -2538,6 +2538,14 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
                 targets.size());
     }
 
+    bool is_immutable(const ASR::ttype_t *type) {
+        if (ASR::is_a<ASR::Character_t>(*type) || ASR::is_a<ASR::Tuple_t>(*type)
+            || ASR::is_a<ASR::Complex_t>(*type)) {
+            return true;
+        }
+        return false;
```

Comment:
```suggestion
        return ((ASR::is_a<ASR::Character_t>(*type) || ASR::is_a<ASR::Tuple_t>(*type) || ASR::is_a<ASR::Complex_t>(*type)));
```

Reply:
Let's put it in `asr_utils.h`. We might need to re-use it in some ASR->ASR passes.